integration-core/src/main/java/org/symphonyoss/integration/core/service/CryptoServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+package org.symphonyoss.integration.core.service;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.symphonyoss.integration.exception.CryptoException;
+import org.symphonyoss.integration.logging.LogMessageSource;
+import org.symphonyoss.integration.service.CryptoService;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.ByteBuffer;
+import java.security.AlgorithmParameters;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.AlgorithmParameterSpec;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.InvalidParameterSpecException;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Implementation of a text-based cryptography service.
+ * Created by campidelli on 9/5/17.
+ */
+@Component
+public class CryptoServiceImpl implements CryptoService {
```

Comment:
I would like to see more javadocs on this class once this is a complex service.

Reply:
Done.